data_platform/generate_features/run_layout.py

```
@@ -0,0 +1,42 @@
+    Path
+        Absolute path to the timestamped feature run directory.
+    """
+    if run_dir_name is None:
+        return feature_storage.create_new_run_dir()
+    requested_path = Path(run_dir_name)
+    if (
+        requested_path.is_absolute()
+        or len(requested_path.parts) != 1
+        or requested_path.name in {".", ".."}
+    ):
+        raise ValueError("run_dir_name must be a single feature run directory name")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)